src/game/objects/general/bridge.c

```diff
@@ -163,13 +177,16 @@ void Bridge_DrawBridgeCeiling(
     if (item->current_anim_state != DOOR_OPEN) {
         return;
     }
-    if (!Bridge_OnDrawBridge(item, z, x)) {
+
+    if (!Bridge_OnDrawBridge(item, x, z)) {
         return;
     }
 
-    if (y > item->pos.y) {
-        *height = item->pos.y + STEP_L;
+    if (y <= item->pos.y || item->pos.y <= *height) {
+        return;
     }
+
+    *height = (int16_t)item->pos.y + STEP_L;
```

Comment:
I don't think this cast is necessary as the variable type is already `int16_t`

Reply:
`item->pos.x/y/z` are `int32_t` I believe. Is the cast unneeded? There are some mismatched types around the codebase similar to this it seems. I just changed this since I was already in the relevant code.

Reply:
Most often yes, type promotion is implicit.